### ISSUE TITLE
Finetuning Model occasionally yields extreme evaluation metrics

### ISSUE THREAD

**Issue by FDSRashid** [UKPLab/sentence-transformers#2365]
I have a strange error. I am following the documentation on training a sentence transformers model from scratch, linked [here](https://www.sbert.net/docs/training/overview.html). In my case, the dataset i am using is in arabic, so i use a Arabic model (ArbertV2). I follow the same preprocessing for my data as the original model.  I provide a binary classification evaluator on a my dataset for the model to validate while training. I train for 5 epochs. What happens is: for the first epoch, evaluation metrics look normal, but by the second epoch, evaluation metrics return extreme values. The recall is very high (99), and the threshold for cosine is literally 1. The model is generating pretty much the same embeddings for any given text, which is what causes all the threshold values for all the metrics to be extreme. This approach has worked for other arabic models, so i am unsure what makes this model so problematic. Or is there a approach i need to do when training models made from scratch?

**Comment by johneckberg** [comment]
@FDSRashid I'm not sure if you're still struggling with this issue, but the documentation linked provides an implementation using CosineSimilarityLoss. Since this loss function only cares about reducing the angle between the embeddings of the input pair, it can be made zero by simply assigning all input sentences to a singular output vector. This could be causing the catastrophic collapse you're encountering during training. Try using [MNR Loss](https://www.sbert.net/docs/package_reference/losses.html#multiplenegativesrankingloss) and see if that helps! Using MNR loss will ensure that the angle between input pairs is minimized but also that the embedding space is used as uniformly as possible, preventing collapse.

**Comment by FDSRashid** [comment]
Thanks, that did the trick!
